flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -236,7 +240,20 @@ public void notifyCheckpointComplete(long checkpointId) {
         // checkpoint coordinator time thread.
         // we can remove the delegation once the checkpoint coordinator runs fully in the
         // scheduler's main thread executor
-        mainThreadExecutor.execute(() -> coordinator.notifyCheckpointComplete(checkpointId));
+        mainThreadExecutor.execute(
+                () -> {
+                    coordinator.notifyCheckpointComplete(checkpointId);
+                    for (Map.Entry<Integer, CloseableSubtaskGateway> entry :
+                            subtaskGatewayMap.entrySet()) {
+                        if (entry.getValue().isClosed()) {
+                            LOG.warn(
```

Comment:
Is there any case that this would happen? If no, would it be simpler to throw exception?

Reply:
I agree. I'll make the change.